runtime/src/account_info.rs

```diff
@@ -7,11 +7,45 @@ use crate::{
 pub type Offset = usize;
 
 /// specify where account data is located
+#[derive(Debug)]
 pub enum StorageLocation {
     AppendVec(AppendVecId, Offset),
     Cached,
 }
 
+impl StorageLocation {
+    pub fn offsets_equal(&self, other: &StorageLocation) -> bool {
```

Comment:
Could the name be made more rusty? These boolean functions tend to be `is_xxx`; would `is_offset_equal()` work for you?

Since we're already calling the fn on `self`, I was thinking th6e singular `offset` would be sufficient, and then that would work well with `is_` instead of going with `are_`. Otherwise, I could see this being a free function too, like `StorageLocation::are_offsets_equal(&lhs, &rhs)`.